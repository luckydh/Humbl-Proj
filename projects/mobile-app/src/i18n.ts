import i18n from "i18next";
import { initReactI18next, useTranslation } from "react-i18next";
import { register } from "timeago.js";
import { es, pt_BR } from "timeago.js/lib/lang";
import HttpApi from "i18next-http-backend";
import LanguageDetector from "i18next-browser-languagedetector";
//import resources from "./locales/index";
// resources: resources,

export type TFnType = ReturnType<typeof useTranslation>["t"];

register("es", es);
register("pt", pt_BR);

i18n
  .use(HttpApi)
  .use(LanguageDetector)
  .use(initReactI18next) // passes i18n down to react-i18next
  .init({
    backend: {
      loadPath: "https://api.locize.app/f4e7a8d0-8642-44bb-ad02-ce1d5554b83c/latest/{{lng}}/{{ns}}",
      addPath: "https://api.locize.app/f4e7a8d0-8642-44bb-ad02-ce1d5554b83c/latest/add/{{lng}}/{{ns}}",
      allowMultiLoading: false,
    },
    //resources,
    //If we don't have a transalation and only have empty string, display fallback
    returnEmptyString: false,
    fallbackLng: "en",
    debug: false,
    ns: "translation",
    keySeparator: false, // we do not use keys in form messages.welcome
    interpolation: {
      escapeValue: false, // react already safes from xss
    },
  });

export default i18n;
