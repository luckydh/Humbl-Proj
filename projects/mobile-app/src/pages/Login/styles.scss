.login-page-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100%;
  width: 100%;
  padding: 20px;

  form {
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 100%;
    max-width: 600px;

    button {
      margin-top: 20px;
    }
  }

  .error-message {
    margin-top: 20px;
    padding: 20px;
    background-color: red;
    text-align: center;
    opacity: 0.7;
    width: 100%;
  }

  .humbl-logo {
    position: absolute;
    top: 10px;
    height: 30px;
  }

  .account-options {
    position: absolute;
    bottom: 50px;
    display: flex;
    flex-direction: column;
    align-items: center;

    &__create {
      margin-top: 10px;
    }

    a {
      text-decoration: none;
      color: white;
    }
  }

  .form-input:not(:first-child) {
    margin-top: 20px;
  }
}


