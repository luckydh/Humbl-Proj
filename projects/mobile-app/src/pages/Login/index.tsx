import { LanguagePicker } from "components/Language/LanguagePicker";
import { Message } from "components/Message/Message";
import { Heading } from "components/Text/Text";
import React, { useState } from "react";
import { useForm } from "react-hook-form";
import { useTranslation } from "react-i18next";
import { Link, useHistory } from "react-router-dom";
import EVENTS from "utils/analytics/AnalyticEvents";
import { identifyUser, trackEvent } from "utils/analytics/Segment";
import { loginUser } from "../../Firebase";
import Button from "components/Button/Button";
import Input from "components/Input/Input";
import "./styles.scss";

type FormInputs = {
  email: string;
  password: string;
};

const initialValues: FormInputs = {
  email: "",
  password: "",
};

const Login: React.FC = () => {
  const { t } = useTranslation();
  const history = useHistory();

  const { handleSubmit, register, errors } = useForm<FormInputs>({
    defaultValues: initialValues,
    mode: "onChange",
  });

  const [hasError, setHasError] = useState<boolean>(false);

  const login = async ({ email, password }: FormInputs) => {
    identifyUser(email);
    loginUser(email, password)
      .then(() => {
        trackEvent(EVENTS.LOGIN_SUCCESS, { status: "success" });
        history.push("/verify");
      })
      .catch((error) => {
        trackEvent(EVENTS.LOGIN_FAILURE, { error });
        setHasError(error);
      });
  };

  return (
    <div className="flex flex-grow flex-col h-full flex-shrink-0">
      <div className="flex flex-col flex-1 flex-shrink-0 justify-center">
        <div className="flex flex-col items-center justify-center">
          <Heading ariaLabel="LOGIN_TITLE_LABEL">{t("login-page.title.login")}</Heading>
        </div>
        <form onSubmit={handleSubmit(login)}>
          {hasError && (
            <div className="my-2">
              <Message variant="error">{t("login-page.form.error.invalid-login")}</Message>
            </div>
          )}
          <div className="my-4">
            <Input
              ariaLabel="LOGIN_EMAIL"
              label={t("login-page.label.email")}
              hasError={!!errors.email}
              placeholder={t("login-page.input.placeholder.email-address")}
              name="email"
              type="email"
              register={register({
                required: {
                  value: true,
                  message: t("login-page.input.email.error.email-is-required"),
                },
              })}
            />
            {errors.email && (
              <div className="my-2">
                <Message variant="error">{errors.email.message}</Message>
              </div>
            )}
          </div>
          <div className="my-4">
            <Input
              label={t("login-page.label.password")}
              ariaLabel="LOGIN_PASSWORD"
              hasError={!!errors.password}
              isSecure
              placeholder={t("login-page.input.placeholder.password")}
              name="password"
              register={register({
                required: {
                  value: true,
                  message: t("login-page.input.password.error.password-is-required"),
                },
              })}
            />
            {errors.password && (
              <div className="my-2">
                <Message variant="error">{errors.password.message}</Message>
              </div>
            )}
          </div>

          <div className="my-6 flex justify-center">
            <Button
              ariaLabel="LOGIN_LOGIN_BUTTON"
              onClick={() => trackEvent(EVENTS.LOGIN_ATTEMPT, { previousPage: "/login" })}
              type="submit">
              {t("login-page.button.login")}
            </Button>
          </div>
          <div className="mb-4 flex flex-col text-xl">
            <Link
              onClick={() => trackEvent(EVENTS.FORGOT_PASSWORD_ATTEMPT, { previousPage: "/login" })}
              aria-label="LOGIN_FORGOTPASSWORD_BUTTON"
              className="mx-auto m-1 text-white"
              to="/forgotpass"
              replace>
              {t("login-page.link.forgot-password")}
            </Link>
            <Link
              onClick={() => trackEvent(EVENTS.SIGN_UP_ATTEMPT, { previousPage: "/login" })}
              to="/signup"
              aria-label="LOGIN_CREATEACCOUNT_BUTTON"
              className="mx-auto m-1 text-white"
              replace>
              {t("login-page.link.create-a-new-account")}
            </Link>
          </div>
        </form>
      </div>
      <div className="mb-4 flex flex-col text-lg">
        <LanguagePicker />
      </div>
    </div>
  );
};

export default Login;
