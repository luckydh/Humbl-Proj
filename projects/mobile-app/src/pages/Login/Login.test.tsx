import React from "react";
import { StaticRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Login from "./index";
//Disable suspense on login
import i18n from "../../i18n";

if (i18n.options) {
  i18n.options.react = {
    useSuspense: false,
  };
}

it("renders Login page", async () => {
  const tree = renderer
    .create(
      <StaticRouter>
        <Login />
      </StaticRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
