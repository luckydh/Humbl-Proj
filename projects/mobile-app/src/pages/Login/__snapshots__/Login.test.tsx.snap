// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Login page 1`] = `
<div
  className="flex flex-grow flex-col h-full flex-shrink-0"
>
  <div
    className="flex flex-col flex-1 flex-shrink-0 justify-center"
  >
    <div
      className="flex flex-col items-center justify-center"
    >
      <h1
        aria-label="LOGIN_TITLE_LABEL"
        className="font-medium text-3xl leading-snug text-white"
      >
        login-page.title.login
      </h1>
    </div>
    <form
      onSubmit={[Function]}
    >
      <div
        className="my-4"
      >
        <div
          className="mb-2"
        >
          <div
            className="ml-2"
          >
            <label
              aria-label="LOGIN_EMAIL_LABEL"
              className="block text-xs uppercase text-white opacity-80 font-medium mb-1"
            >
              login-page.label.email
            </label>
          </div>
          <div
            className="py-0 w-full flex items-center rounded-lg select-none bg-blue border-white border-2 border-solid outline-none"
          >
            <input
              className="text-white placeholder-white-faded w-full rounded-lg bg-transparent text-md sm:text-xl outline-none border-none focus:border-transparent focus:outline-none focus:ring-0 focus:placeholder-transparent"
              name="email"
              placeholder="login-page.input.placeholder.email-address"
              style={
                Object {
                  "fontSize": 20,
                  "height": 48,
                }
              }
              type="email"
            />
          </div>
        </div>
      </div>
      <div
        className="my-4"
      >
        <div
          className="mb-2"
        >
          <div
            className="ml-2"
          >
            <label
              aria-label="LOGIN_PASSWORD_LABEL"
              className="block text-xs uppercase text-white opacity-80 font-medium mb-1"
            >
              login-page.label.password
            </label>
          </div>
          <div
            className="py-0 w-full flex items-center rounded-lg select-none bg-blue border-white border-2 border-solid outline-none"
          >
            <input
              className="text-white placeholder-white-faded w-full rounded-lg bg-transparent text-md sm:text-xl outline-none border-none focus:border-transparent focus:outline-none focus:ring-0 focus:placeholder-transparent"
              name="password"
              placeholder="login-page.input.placeholder.password"
              style={
                Object {
                  "fontSize": 20,
                  "height": 48,
                }
              }
              type="password"
            />
            <div
              className="max-h-8 flex"
              style={
                Object {
                  "maxWidth": "3.5rem",
                }
              }
            >
              <div
                aria-label="LOGIN_PASSWORD_SHOW_INPUT"
                className="text-xs mr-2"
                onClick={[Function]}
              >
                input.show
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="my-6 flex justify-center"
      >
        <button
          aria-label="LOGIN_LOGIN_BUTTON"
          className="text-white select-none outline-none flex justify-center items-center justify-center font-medium w-full text-white py-2 px-4 bg-blue-dark rounded-md active:bg-blue-light active:shadow-button-active active:text-blue-dark transition-all py-3 px-6 text-md sm:leading-none"
          onClick={[Function]}
          style={
            Object {
              "minWidth": 160,
            }
          }
          type="submit"
        >
          login-page.button.login
        </button>
      </div>
      <div
        className="mb-4 flex flex-col text-xl"
      >
        <a
          aria-label="LOGIN_FORGOTPASSWORD_BUTTON"
          className="mx-auto m-1 text-white"
          href="/forgotpass"
          onClick={[Function]}
        >
          login-page.link.forgot-password
        </a>
        <a
          aria-label="LOGIN_CREATEACCOUNT_BUTTON"
          className="mx-auto m-1 text-white"
          href="/signup"
          onClick={[Function]}
        >
          login-page.link.create-a-new-account
        </a>
      </div>
    </form>
  </div>
  <div
    className="mb-4 flex flex-col text-lg"
  >
    <button
      aria-label="CHANGELANGUAGE_BUTTON"
      className="mx-auto m-1 text-white text-xl flex flex-col items-center"
      onClick={[Function]}
      type="button"
    >
      <ion-icon
        icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Globe</title><path d='M256 48C141.13 48 48 141.13 48 256s93.13 208 208 208 208-93.13 208-208S370.87 48 256 48z' stroke-miterlimit='10' class='ionicon-fill-none ionicon-stroke-width'/><path d='M256 48c-58.07 0-112.67 93.13-112.67 208S197.93 464 256 464s112.67-93.13 112.67-208S314.07 48 256 48z' stroke-miterlimit='10' class='ionicon-fill-none ionicon-stroke-width'/><path d='M117.33 117.33c38.24 27.15 86.38 43.34 138.67 43.34s100.43-16.19 138.67-43.34M394.67 394.67c-38.24-27.15-86.38-43.34-138.67-43.34s-100.43 16.19-138.67 43.34' stroke-linecap='round' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/><path stroke-miterlimit='10' d='M256 48v416M464 256H48' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
      />
      <span
        className="text-base"
      >
        core.link.language
      </span>
    </button>
  </div>
</div>
`;
